packages/block-library/src/avatar/edit.js

```diff
@@ -0,0 +1,222 @@
+				onResizeStop={ ( event, direction, elt, delta ) => {
+					setAttributes( {
+						size: parseInt(
+							attributes.size + ( delta.height || delta.width ),
+							10
+						),
+					} );
+				} }
+				lockAspectRatio
+				enable={ {
+					top: false,
+					right: ! isRTL(),
+					bottom: true,
+					left: isRTL(),
+				} }
+				minWidth={ avatar.minSize }
+				maxWidth={ avatar.maxSize }
+			>
+				<img
+					src={ avatar.src }
+					alt={ avatar.alt }
+					{ ...borderProps }
+					className={ classnames(
+						'avatar',
+						'avatar-' + attributes.size,
+						'photo',
+						'wp-block-avatar__image',
+						borderProps.className
+					) }
+					style={ {
+						...borderProps.style, // Border radius, width and style.
+					} }
+				/>
+			</ResizableBox>
+		</div>
+	);
+};
+const CommentEdit = ( { attributes, context, setAttributes, isSelected } ) => {
+	const { commentId } = context;
+	const blockProps = useBlockProps();
+	const avatar = useCommentAvatar( { commentId } );
+	return (
+		<>
+			<AvatarInspectorControls
+				avatar={ avatar }
+				setAttributes={ setAttributes }
+				attributes={ attributes }
+				selectUser={ false }
+			/>
+			{ attributes.isLink ? (
+				<a
+					href="#avatar-pseudo-link"
+					className="wp-block-avatar__link"
+					onClick={ ( event ) => event.preventDefault() }
+				>
+					<ResizableAvatar
+						attributes={ attributes }
+						avatar={ avatar }
+						blockProps={ blockProps }
+						isSelected={ isSelected }
+						setAttributes={ setAttributes }
+					/>
+				</a>
+			) : (
+				<ResizableAvatar
+					attributes={ attributes }
+					avatar={ avatar }
+					blockProps={ blockProps }
+					isSelected={ isSelected }
+					setAttributes={ setAttributes }
+				/>
+			) }
+		</>
+	);
+};
+
+const UserEdit = ( { attributes, context, setAttributes, isSelected } ) => {
+	const { postId, postType } = context;
+	const avatar = useUserAvatar( {
+		userId: attributes?.userId,
+		postId,
+		postType,
+	} );
+	const blockProps = useBlockProps();
+	return (
+		<>
+			<AvatarInspectorControls
+				selectUser={ true }
+				attributes={ attributes }
+				avatar={ avatar }
+				setAttributes={ setAttributes }
+			/>
+			<div>
+				{ attributes.isLink ? (
+					<a
+						href="#avatar-pseudo-link"
+						className="wp-block-avatar__link"
+						onClick={ ( event ) => event.preventDefault() }
+					>
+						<ResizableAvatar
+							attributes={ attributes }
+							avatar={ avatar }
+							blockProps={ blockProps }
+							isSelected={ isSelected }
+							setAttributes={ setAttributes }
+						/>
+					</a>
+				) : (
+					<ResizableAvatar
+						attributes={ attributes }
+						avatar={ avatar }
+						blockProps={ blockProps }
+						isSelected={ isSelected }
+						setAttributes={ setAttributes }
+					/>
+				) }
+			</div>
+		</>
+	);
+};
+
+export default function Edit( props ) {
+	if ( props?.context?.commentId ) {
+		return <CommentEdit { ...props } />;
+	}
+	return <UserEdit { ...props } />;
```

Comment:
> How about we extract the shared part into an `AvatarEdit` component, and move the avatar determination logic into `Edit`?

Hey @ockham, for me that would be the ideal solution. 😄  However, I guess @c4rl0sbr4v0's decision of making two different components here was because React hooks cannot be used conditionally. 🤷 

Another option that just came to my mind could be moving the conditional checking inside a `useAvatar` hook, maybe?

Reply:
The Comment Avatar and the User Avatar have different hooks to retrieve the data. So adding a conditional on `useAvatar` function would cause the React hooks error 😢 

![Screenshot 2022-03-17 at 11 21 23](https://user-images.githubusercontent.com/37012961/158788694-bceb30ed-153b-4252-a124-c6aac10b260f.png)